lib/client.js

```diff
@@ -901,6 +905,16 @@ function onSocketConnect () {
 }
 
 function onSocketError (err) {
+  const { [kParser]: parser } = this
+  if (err.code === 'ECONNRESET' && parser.statusCode && !parser.shouldKeepAlive) {
+    // Response does not contain content-length nor chunked encoding.
+    // We treat all incoming data so for as a valid response.
+    currentParser = parser
+    parser.finish()
+    currentParser = null
```

Comment:
Cna we move the currentParser stuff into `parser.finish()`?

Reply:
I've followed the same pattern we are using elsewhere, we set `currentParser` outside of the parser class.

Reply:
No we don't? We set it inside `execute` which is part of the parser class.